### DISCUSSION TITLE
Parsing multiple log formats

### DISCUSSION THREAD

**Discussion by opicodes** [opcodesio/log-viewer#111] [Q&A] [answered]
In the config file I see a *patterns* key (with a laravel value that matches laravel logs).

Being patterns plural, and an array, I suppose is possible to define multiple patters to match different log file formats (e.g. I'd need it to visualise supervisord logs that, at the moment, obviously render a blank page that says "No results".

Unfortunately there's no docs for this configuration and I can't quite figure it out. Could anyone help or at least point me in the right direction please?

Thanks

**Comment by arukompas** [comment, marked as answer]
Hey @opicodes 

This is done to future-proof the configuration file so there's as little conflicts as possible later on.

At the moment there's no way to define additional log formats to parse. But it's definitely in the backlog :)

**Reply by opicodes** [reply]
I see...

If you're open to pull requests and want to have a chat about it to choose a possible implementation, I'd be happy to give it a go in my spare time.

**Reply by arukompas** [reply]
Hey @opicodes - thanks for the offer. You can submit a PR if you want, but I can't promise it'll be merged. This is a very important feature and I want it done right. I'm not sure whether to go the class route, config route, or defining it inside your AppServiceProvider. I know that I want the package to have a lot of default log readers (such as apache, mysql, redis, cron, etc) which could be contributed easily by the community. So it's important to get the interface right so that any additional log readers can be added very easily.

**Reply by opicodes** [reply]
I agree, that's why I proposed to have a chat first. Maybe you could describe what's already in place and could be reused, and/or what needs to be added. Personally I think the config route would be best, reusing the "patterns" already there, also I thinkl it wouldn't be particularly complex.
